src/rred_reports/reports/auth.py

```diff
@@ -0,0 +1,32 @@
+"""Mail server authentication"""
+from dataclasses import dataclass
+
+from dynaconf.base import Settings
+from exchangelib import (
+    DELEGATE,
+    Account,
+    Configuration,
+    OAuth2LegacyCredentials,
+)
+
+from rred_reports.reports import get_settings
+
+
+@dataclass
+class RREDAuthenticator:
+    settings: Settings = get_settings()
+
+    def get_credentials(self) -> OAuth2LegacyCredentials:
+        return OAuth2LegacyCredentials(
+            client_id=self.settings.client_id,
+            client_secret=self.settings.client_secret,
+            tenant_id=self.settings.tenant_id,
+            username=f"{self.settings.username}@ucl.ac.uk",
+            password=self.settings.auth_pass,
+        )
```

Comment:
so much fun, also worth adding some instructions to the readme for setting up the application in azure.  Imagine this will need to be per user? Doesn't need to be super detailed but pointing to an existing guide would be great. maybe a  `Configuration for emailing` section?